File: Daily/PY/Leetcode279-完全平方数.py
class Solution:
    def numSquares(self, n: int) -> int:
        dp = [n] * (n + 1)
        dp[0] = 0
        i = 1
        while i * i <= n:
            for j in range(i * i,n + 1):
                dp[j] = min(dp[j],dp[j - i * i] + 1)
            i += 1
        return dp[n]

File: Leetcode/Rust/Leetcode279-完全平方数.rs
impl Solution {
    pub fn num_squares(n: i32) -> i32 {
        let n = n as usize;
        let mut dp = vec![n as i32;n + 1];
        dp[0] = 0;
        for i in 1..=n{
            let mut j = 1;
            while j * j <= i{
                dp[i] = dp[i].min(dp[i - j * j] + 1);
                j += 1
            }
        }
        dp[n]
    }
}